gdb-server/src/arch.rs

```diff
@@ -0,0 +1,61 @@
+use gdbstub::arch::{Arch, RegId, Registers, SingleStepGdbBehavior};
+
+// Placeholder type for runtime architecture trait data
+pub(crate) enum RuntimeArch {}
+
+/// Generic implementation for a runtime evaluated architecture
+impl Arch for RuntimeArch {
+    // We can always handle a 64-bit address space
```

Comment:
This is just for the internal types used by the stub.  GDB knows the architecture so it'll bound any input / output to the right size for the pc register so it will never ask for something out of bounds.